src/components/activity-page/managed-interactive/managed-interactive.tsx

```diff
@@ -81,44 +88,72 @@ export const ManagedInteractive: React.FC<IProps> = (props) => {
       setHint(newHint);
     }, []);
 
+    const embeddableAuthoredState = () => {
+      // enable modal support for activity player only
+      const parsedState = embeddable.authored_state ? JSON.parse(embeddable.authored_state) : {};
+      parsedState.modalSupported = true;
+      return JSON.stringify(parsedState);
+    };
+
+    const getModalContainer = (): HTMLElement => {
+      return document.getElementById("app") || document.body;
+    };
+
+    const toggleModal = useCallback(() => {
+      setShowModal(!showModal);
+    }, [showModal]);
+
+    const handleCloseModalRequest = (modalProps?: any) => {
+      if (modalProps) {
+        handleNewInteractiveState(modalProps.interactiveState);
+      }
+    };
```

Comment:
@dstrawberrygirl, do you remember what was the intention here? How would we get modalProps with interactiveState here? I'm going to change this function, as currently, it's impossible to close the modal, but I'm double-checking if I'm not breaking something.

Reply:
I agree that this is hairy, but it does appear to work in its current form (modal can be opened/closed, unless I'm missing something).  See this example:
https://activity-player.concord.org/branch/master/?preview&activity=sample-question-interactive-complex&page=1

Reply:
I can't actually recall the details here... I suspect I may have added the possibility of modalProps as a placeholder based on a previous implementation. Sorry I don't have anything more concrete to add, you should probably go ahead and make sure to test in both activity player and LARA to check for breakages.

Reply:
> I agree that this is hairy, but it does appear to work in its current form (modal can be opened/closed, unless I'm missing something). See this example:

Actually modal can't be closed when you click outside modal borders. This function needs to do that.
I think I figured it out. `modalProps` argument can be removed, as it will be never passed to this function. It's pretty random here, maybe it got lost during some refactoring. It's a function called automatically when you click modal background. So, it should just close the modal and that's all. I just wasn't sure if I'm not breaking something as `handleNewInteractiveState(modalProps.interactiveState);` sounded serious, but it doesn't seem so. ;)